### PR TITLE
MNIST example: use regular dropout rather than dropout2d

### PR DESCRIPTION
**Pull request by eli-osherovich**
Using dropout2d does not really make sense for the second dropout since the data is flattened at that point (but it works accidentally). Furthermore, since this example copies the one from Keras (https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py) it might be better to follow that example (regular dropout). 

No noticeable performance changes.

### CODE REVIEW

**Comment by soumith** [comment]
the example started as a migration from torch7, but at some point, someone must've contributed a patch to mimic the Keras model.

**Comment by soumith** [comment]
looks like https://github.com/pytorch/examples/pull/658 was where we moved over to the Keras-example style model
